### ISSUE TITLE
`extractCss` adds unclosed `link` tag, breaks XHTML index pages

### ISSUE THREAD

**Issue by MilesBHuff** [angular/angular-cli#16964]
# Feature

### Command (mark with an `x`)

This is a Webpack configuration issue, and is not directly related to any of Angular CLI's commands.

### Is this a regression?

Yes (fixed in 2016; broken at some later date) (https://github.com/angular/angular-cli/pull/3743)

### Description

The `extractCss` setting in `angular.json` adds an unclosed `link` tag to the DOM.  This breaks XHTML index pages.

## 🔬 Minimal Reproduction

1. In `angular.json`, set `extractCss` to `true` for all builds.
2. Make sure your `index.xhtml` file is valid polyglot HTML5/XML.
3. Start the dev server.

## 🔥 Exception or Error

    XML Parsing Error: mismatched tag. Expected: </link>.
    Location: http://localhost:4200/
    Line Number ##, Column 45:
        <link rel="stylesheet" href="styles.css"></head>

## 🌍  Your Environment

**Angular Version:** `9.0.1`

**Anything else relevant?** No, this issue's pretty simple.

**Comment by MilesBHuff** [comment]
Moved from https://github.com/angular/angular/issues/35404.

**Comment by MilesBHuff** [comment]
This issue was fixed in 2016, but seems to have recurred.  (https://github.com/angular/angular-cli/issues/3217)

**Comment by santoshyadav198613** [comment]
You can read here https://en.m.wikipedia.org/wiki/Polyglot_markup
![Screenshot_20200213-123844](https://user-images.githubusercontent.com/11923975/74409922-d757c500-4e5d-11ea-82f4-506270243a9d.jpg)

**Comment by santoshyadav198613** [comment]
Sorry for confusion earlier , what I wanted to write is in polygot all tags should have closing tag.

**Comment by dgp1130** [comment]
We don't officially support XHTML at the moment, so this isn't exactly a regression. It may have worked in the past, but that is more coincidental rather than intentional. Proper support for XHTML would include full test coverage across the toolchain to verify XHTML support.

This is more of a general feature request. @IgorMinar, any thoughts of whether we want to take the effort to fully support XHTML?

**Comment by alan-agius4** [comment]
Note: Ivy compiler will generate HTML5 code. If you insert XHTML in a template ex: `<br />`, the rendered HTML will be HTML `<br _ngcontent-rjm-c0>`.

**Comment by MilesBHuff** [comment]
@alan-agius4 It actually isn't a feature request, and is indeed a regression:  XHTML support here was deliberately implemented, as can be seen in the issue I linked earlier:  https://github.com/angular/angular-cli/issues/3217#issuecomment-269234599

**Comment by sacgrover** [comment]
As per my understanding, these link tags are not a part of Ivy compiler as they are included in index.html not generated by any component.  W3 Validator will validate only index.html static tags not the element created dynamically.  So closing the link tag should resolve this issue.

**Comment by MilesBHuff** [comment]
That is correct, the link tags are completely outside of Ivy.  The fix is a simple boolean setting in a Webpack plugin.

**Comment by IgorMinar** [comment]
After looking briefly through this and related issues I agree that we should treat this as a regression and add on integration test to prevent further regressions.

**Comment by dgp1130** [comment]
I filed a separate issue for adding comprehensive test coverage and making sure we don't regress from future changes and verify our current support across the toolchain. That's a non-trivial amount of work, so we'll need to budget for it accordingly: https://github.com/angular/angular-cli/issues/17036.

This issue can just worry about the individual `<link>` tag, and we can address any other regressions we find as they come up organically or as we identify them through adding these tests.

**Comment by dgp1130** [comment]
We were looking at old PRs and decided to close https://github.com/angular/angular-cli/pull/16965#issuecomment-829501778, as there are some open questions about how exactly this will work and it's actually blocked on XHTML support in the compiler, which requires a separate effort.

In the previous comment we decided to scope this issue just to the `<link>` tag, but that isn't easily fixable on its own without pulling on a lot of other questions around general XHTML support (for example: Should we have an `--xhtml` flag and if so should we add it before we have full support of XHTML across all of Angular?).

For now, we think the best solution is to close this particular issue around the `<link>` tag and follow up with the broader XHTML discussion in https://github.com/angular/angular-cli/issues/18044. I'll bring this up in the relevant planning discussions to get a stronger direction and find the right priority for this work. Hopefully we'll have more updates to share on that issue soon.

**Comment by MilesBHuff** [comment]
Thanks for the update, @dgp1130!
